Sources/web3swift/Web3/Web3+Structures.swift

```diff
@@ -52,13 +52,13 @@ extension TransactionOptions: Decodable {
     
     public init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
-        if let gasLimit = try decodeHexToBigUInt(container, key: .gas) {
+        if let gasLimit = try decodeHexToBigUInt(container, key: .gas, allowOptional:true) {
```

Comment:
the problem here is that to code expects to drop down to provide a default if the call to `decodeHexToBigUInt` returns with nil. `decodeHexToBigUInt` will `throw Web3Error.dataError` instead of `return nil` unless `allowOptional` is set to true. So this `if let` is meaningless without the `allowOptional: true`.

On a larger scope, yes nonce should never be optional, I was not trying to address that issue. That should be handled at a higher level. The input here could come from a number of sources, where nonce may no be known yet. The same goes for the various gas values, as if this is a 1559 transaction those values may not exist. Furthermore, if the transaction is pending and not mined yet, those values may also not be present, as they are not known yet. Issue #469